docs/user-guide/logs/manage-pipelines.md

```diff
@@ -87,4 +87,253 @@ Output:
+            "name": "id1"
+        },
+        {
+            "colume_type": "FIELD",
+            "data_type": "INT32",
+            "fulltext": false,
+            "name": "id2"
+        },
+        {
+            "colume_type": "FIELD",
+            "data_type": "STRING",
+            "fulltext": false,
+            "name": "type"
+        },
+        {
+            "colume_type": "FIELD",
+            "data_type": "STRING",
+            "fulltext": false,
+            "name": "log"
+        },
+        {
+            "colume_type": "FIELD",
+            "data_type": "STRING",
+            "fulltext": false,
+            "name": "logger"
+        },
+        {
+            "colume_type": "TIMESTAMP",
+            "data_type": "TIMESTAMP_NANOSECOND",
+            "fulltext": false,
+            "name": "time"
+        }
+    ]
+}
+```
+
+The output shows that the pipeline successfully processed the log data. The `rows` field contains the processed data, and the `schema` field contains the schema information of the processed data. You can use this information to verify the correctness of the pipeline configuration.
+
+### Test a Failed Pipeline
+
+Assume that the pipeline configuration is as follows:
+
+
+```bash
+curl -X "POST" "http://localhost:4000/v1/events/pipelines/test" \
+     -H 'Content-Type: application/x-yaml' \
+     -d $'processors:
+  - date:
+      field: time
+      formats:
+        - "%Y-%m-%d %H:%M:%S%.3f"
+      ignore_missing: true
+  - gsub:
+      fields:
+        - message
+      pattern: "\\\."
+      replacement:
+        - "-"
+      ignore_missing: true
+
+transform:
+  - fields:
+      - message
+    type: string
+  - field: time
+    type: time
+    index: timestamp'
+```
+
+The pipeline configuration contains an error. The `gsub` Processor expects the `replacement` field to be a string, but the current configuration provides an array. As a result, the pipeline creation fails with the following error message:
+
+
+```json
+{"error":"Failed to parse pipeline: 'replacement' must be a string"}
+```
+
+Therefore, We need to modify the configuration of the `gsub` Processor and change the value of the `replacement` field to a string type.
+
+```bash
+curl -X "POST" "http://localhost:4000/v1/events/pipelines/test" \
+     -H 'Content-Type: application/x-yaml' \
+     -d $'processors:
+  - date:
+      field: time
+      formats:
+        - "%Y-%m-%d %H:%M:%S%.3f"
+      ignore_missing: true
+  - gsub:
+      fields:
+        - message
+      pattern: "\\\."
+      replacement: "-"
+      ignore_missing: true
+
+transform:
+  - fields:
+      - message
+    type: string
+  - field: time
+    type: time
+    index: timestamp'
+```
+
+The Pipeline has been successfully created at this point, and We can test the Pipeline using the `dryrun` interface. We will test it with erroneous log data where the value of the message field is in numeric format, causing the pipeline to fail during processing.
```

Comment:
```suggestion
The Pipeline has been successfully updated at this point, and We can test the Pipeline using the `dryrun` interface. We will test it with erroneous log data where the value of the message field is in numeric format, causing the pipeline to fail during processing.
```

Reply:
Since an error was thrown during the first creating request, the pipeline does not have been successfully created the first time? So it should be created here?

Reply:
what is the `dryrun` interface?

Reply:
Oh yes the first one is not created. My bad :P

> what is the dryrun interface?

`dryrun` typically means run with given configuration, get the output, but **does not** actually take effect. In this case, persist the record into database.